code/modules/antagonists/heretic/influences.dm

```diff
@@ -64,28 +105,149 @@
  */
-/datum/reality_smash_tracker/proc/generate_new_influences()
-	// 1 heretic = 4 influences
-	// 2 heretics = 7 influences
-	// 3 heretics = 9 influences
-	// 4 heretics = 10 influences, +1 for each onwards.
-	var/how_many_can_we_make = 0
-	for(var/heretic_number in 1 to length(tracked_heretics))
-		how_many_can_we_make += max(NUM_INFLUENCES_PER_HERETIC - heretic_number + 1, 1)
+/datum/reality_smash_tracker/proc/generate_new_influences(amount)
+	/// Static list of station areas which would otherwise be valid, but we don't want to spawn influences in.
+	var/static/list/station_area_blacklist = typecacheof(list(
+		/area/chapel/office,
+		/area/commons/dorms/barracks,
+		/area/crew_quarters/theatre/abandoned,
+		/area/crew_quarters/theatre/backstage,
+		/area/hallway/secondary/asteroid,
+		/area/hallway/secondary/command,
+		/area/hallway/secondary/construction,
+		/area/hallway/secondary/service,
+		/area/hallway/upper/secondary/command,
+		/area/hallway/upper/secondary/construction,
+		/area/hallway/upper/secondary/service,
+		/area/holodeck/rec_center/offstation_one,
+		/area/hydroponics/garden/abandoned,
+		/area/library/abandoned
+	))
+	/// Static list of station areas we will attempt to spawn influences in the general vicinity of.
+	var/static/list/public_station_areas = typecacheof(list(
+		/area/cargo/lobby,
+		/area/chapel,
+		/area/commons/dorms,
+		/area/crew_quarters/cafeteria,
+		/area/crew_quarters/fitness,
+		/area/crew_quarters/locker,
+		/area/crew_quarters/lounge,
+		/area/crew_quarters/park,
+		/area/crew_quarters/theatre,
+		/area/crew_quarters/toilet,
+		/area/hallway,
+		/area/holodeck/rec_center,
+		/area/hydroponics/garden,
+		/area/library,
+		/area/medical/medbay/central,
+		/area/medical/medbay/lobby,
+		/area/medical/sleeper,
+		/area/science/lobby,
+		/area/storage/art,
+		/area/storage/primary,
+		/area/storage/tools
+	)) - station_area_blacklist
+	/// Static typecache of 'secondary' areas that should be weighted less.
+	var/static/list/low_weight_areas = typecacheof(/area/security/checkpoint)
+	/// Static typecache of the areas we will NEVER spawn influences in.
+	var/static/list/forbidden_influence_areas = (typecacheof(list(
+		/area/ai_monitored,
+		/area/asteroid,
+		/area/bridge,
+		/area/comms,
+		/area/crew_quarters/heads,
+		/area/docking,
+		/area/drydock,
+		/area/engine/atmospherics_engine,
+		/area/engine/engine_room,
+		/area/engine/gravity_generator,
+		/area/engine/transit_tube,
+		/area/gateway,
+		/area/maintenance,
+		/area/medical/abandoned,
+		/area/quartermaster/qm,
+		/area/quartermaster/qm_bedroom,
+		/area/science/mixing/chamber,
+		/area/security,
+		/area/server,
+		/area/shuttle,
+		/area/solar,
+		/area/space,
+		/area/tcommsat,
+		/area/teleporter
+	)) + station_area_blacklist) - low_weight_areas
+
+	if(!isnum_safe(amount) || amount <= 0)
+		// 1 heretic = 4 influences
+		// 2 heretics = 7 influences
+		// 3 heretics = 9 influences
+		// 4 heretics = 10 influences, +1 for each onwards.
```

Comment:
Considering that they can now find these easilly and they will almost always be in public areas, this seems like an insane amount of influences to spawn. Surely with 1 heretic you almost have enough to reach ascension level on influences alone without doing any sacrafice.
I also think that if they have something that will tell them where the nearest rift is, then we don't need to check at all if the rift is spawning in a location visible to a public hallway. I would honestly prefer if they would almost always spawn somewhere that you at least have to do some hacking/tailgating to get into